### PR TITLE
ports/mimxrt10xx/boards: Add board files for Seeed ARch Mix, Olimex RT1011 and Makerdiary RT1011 Nano Kit..

### PR DESCRIPTION
**Pull request by robert-hh**
Boards:

- Seed Arch Mix RT1052
- Olimex RT1011
- Makerdiary RT1011 Nano Kit

All sets differ only slightly from the respective imxrt10xx-evk files. Major changes are flash size and LED pin. All have been tested with theses boards.

### CODE REVIEW

File: ports/mimxrt10xx/boards/makerdiary_rt1011/board.h [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+// USB UF2
+//--------------------------------------------------------------------+
+
+#define USB_VID           0x2886
+#define USB_PID           0xf00f
+#define USB_MANUFACTURER  "Makerdiary"
+#define USB_PRODUCT       "iMX RT1011 Nano Kit"
+
+#define UF2_PRODUCT_NAME  USB_MANUFACTURER " " USB_PRODUCT
+#define UF2_BOARD_ID      "iMX RT1011 Nano Kit"
```
Comment (hathach) [inline ports/mimxrt10xx/boards/makerdiary_rt1011/board.h:87]: should follow the board-id format https://github.com/microsoft/uf2?tab=readme-ov-file#files-exposed-by-bootloaders

```
UF2 Bootloader v1.1.3 SFA
Model: Arduino Zero
Board-ID: SAMD21G18A-Zero-v0
```

The Board-ID field is machine-readable and consists of a number of dash-separated tokens. The first token is the CPU type, second is the board type, and third is the board revision. More tokens can be also added.

File: ports/mimxrt10xx/boards/olimex_rt1010/board.h [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+// USB UF2
+//--------------------------------------------------------------------+
+
+#define USB_VID           0x15ba
+#define USB_PID           0x0046
+#define USB_MANUFACTURER  "Olimex"
+#define USB_PRODUCT       "RT1010"
+
+#define UF2_PRODUCT_NAME  USB_MANUFACTURER " " USB_PRODUCT
+#define UF2_BOARD_ID      "Olimex-RT1010"
```
Comment (hathach) [inline ports/mimxrt10xx/boards/olimex_rt1010/board.h:87]: see above board id format

**Review by hathach — COMMENTED** [review]
Thank you for your PR, everything look goods, there is a minor request to change board-id to match uf2 specs format.

Also I have a question regarding VID/PID, are these assigned specifically for these board for bootloader purpose and re-use/shared with other application/board.

File: ports/mimxrt10xx/boards/makerdiary_rt1011/board.h (hunk 1 above)
Comment (robert-hh) [reply ports/mimxrt10xx/boards/makerdiary_rt1011/board.h:87]: Thank you for the hint about the board ID structure. I'll change them accordingly.

**Comment by robert-hh** [comment]
The VID/PID are set on best knowdledge. 
Olimex RT1011: PID and VID have been supplied by the vendor and are set by the APP as well.
Makerdiary RT1010 Nano Kit: The VID/PID are the values shown in their bootstrap loader. So I use the same here, even if the VID is one of Seeed Technologies.
Seeed Arch Mix:  0x2886 is tve VID of Seeed. No specific PID was supplied by the vendor. But I can ask.

**Comment by hathach** [comment]
> The VID/PID are set on best knowdledge. Olimex RT1011: PID and VID have been supplied by the vendor and are set by the APP as well. Makerdiary RT1010 Nano Kit: The VID/PID are the values shown in their bootstrap loader. So I use the same here, even if the VID is one of Seeed Technologies. Seeed Arch Mix: 0x2886 is tve VID of Seeed. No specific PID was supplied by the vendor. But I can ask.

I guess olimex and makerdiary is OK, for Seed Arch, maybe if you could just ask to see if they allocated any PID for the board (you can @ their developer here know one).

**Comment by gsteiert** [comment]
I requested NXP set aside a VID/PID that could be used for generically for
UF2. I would suggest using that for UF2 on NXP based boards. Otherwise you
always need two PID if your application has a different USB configuration
than UF2.

Greg

**Comment by robert-hh** [comment]
@hathach Seeed responded and told me, that they cannot tell me a PID for the Arch Mix board. Maybe they never assigned one, or they could not find the number. The board I have is not any more in the delivery state, so I cannot just check. Anyhow I've set it to 0x0010. Maybe 0x1052 would be a good number as well.

**Comment by hathach** [comment]
> I requested NXP set aside a VID/PID that could be used for generically for UF2. I would suggest using that for UF2 on NXP based boards. Otherwise you always need two PID if your application has a different USB configuration than UF2. Greg

That sounds great, do you think that makes sense to ask for a PID per mcu since they are a bit different from each other. Though I think a general one for imxrt is OK as well for board that does not has a private PID. Btw, @gsteiert since you are here, do you happen to know if there is any issue with imxrt1015, robert only get it working when compiled with gcc v10 https://github.com/adafruit/tinyuf2/discussions/413#discussioncomment-10853224 . I forgot the detail, but could look at this later if you don't have time.

> @hathach Seeed responded and told me, that they cannot tell me a PID for the Arch Mix board. Maybe they never assigned one, or they could not find the number. The board I have is not any more in the delivery state, so I cannot just check. Anyhow I've set it to 0x0010. Maybe 0x1052 would be a good number as well.

The rule of thumb is making sure that the VID/PID is valid and unique if possible (not conflicting with other existing device whose vid/pid is owned by other). If your PID for seed it not used elsewhere then I think we are good with since I think seed agree for you to use their VID. Otherwise we can also wait for greg (FAE from NXP) to see if he could get an generic PID from NXP.

File: ports/mimxrt10xx/port.mk [hunk 3]
```diff
@@ -1,6 +1,7 @@
 UF2_FAMILY_ID = 0x4fb2d5bd
 CROSS_COMPILE = arm-none-eabi-
 
+# SDK_DIR = /home/robert/Downloads/mcux-sdk
```
Comment (hathach) [inline ports/mimxrt10xx/port.mk:4]: probably need to remove this

**Review by hathach — COMMENTED** [review]
there seems a typo in port.mk

File: ports/mimxrt10xx/port.mk (hunk 3 above)
Comment (robert-hh) [reply ports/mimxrt10xx/port.mk:4]: Indeed, it is. I'II remove it.

Comment (robert-hh) [reply ports/mimxrt10xx/port.mk]: Updated.

**Review by hathach — APPROVED** [review]
perfect, thank you

**Comment by hathach** [comment]
you're welcome, I am still stuck with dwc2 host driver for the time being. Will test out the imxrt1015 later on when possible

**Pull request merged**